src/pages/test-report/components/toolbar/toolbar.ts

```diff
@@ -55,15 +62,30 @@ export class ToolbarComponent {
       this.subscription.unsubscribe();
     }
   }
+  toggleremoveFaultMode(): void {
+    if (this.isDangerousMode) {
+      this.store$.dispatch(new ToggleDangerousFaultMode());
+    }
+    if (this.isSeriousMode) {
+      this.store$.dispatch(new ToggleSeriousFaultMode());
+    }
+    this.store$.dispatch(new ToggleRemoveFaultMode());
+  }
 
   toggleSeriousMode(): void {
+    if (this.isRemoveFaultMode) {
+      this.store$.dispatch(new ToggleRemoveFaultMode());
+    }
     if (this.isDangerousMode) {
       this.store$.dispatch(new ToggleDangerousFaultMode());
     }
     this.store$.dispatch(new ToggleSeriousFaultMode());
   }
 
   toggleDangerousMode(): void {
+    if (this.isRemoveFaultMode) {
+      this.store$.dispatch(new ToggleRemoveFaultMode());
+    }
     this.store$.dispatch(new ToggleDangerousFaultMode());
   }
 
```

Comment:
should this not also toggle Serious mode if it's set.... if it's about turning off the state of the other two buttons.

Reply:
Currently as the design stands you can't turn on dangerous mode while you're in serious mode. But probably worth putting in the guard to this function.